src/overlays/actors/ovl_Boss_Fd/z_boss_fd.c

```diff
@@ -23,275 +57,1936 @@ const ActorInit Boss_Fd_InitVars = {
+}
+
+s32 BossFd_IsFacingLink(BossFd* this) {
+    return ABS((s16)(this->actor.yawTowardsPlayer - this->actor.world.rot.y)) < 0x2000;
+}
+
+void BossFd_SetupFly(BossFd* this, GlobalContext* globalCtx) {
+    Animation_PlayOnce(&this->skelAnimeHead, &gVolvagiaHeadEmergeAnim);
+    Animation_PlayOnce(&this->skelAnimeRightArm, &gVolvagiaRightArmEmergeAnim);
+    Animation_PlayOnce(&this->skelAnimeLeftArm, &gVolvagiaLeftArmEmergeAnim);
+    this->actionFunc = BossFd_Fly;
+    this->fwork[BFD_TURN_RATE_MAX] = 1000.0f;
+}
+
+static Vec3f sHoleLocations[] = {
+    { 0.0f, 90.0f, -243.0f },    { 0.0f, 90.0f, 0.0f },    { 0.0f, 90.0f, 243.0f },
+    { -243.0f, 90.0f, -243.0f }, { -243.0f, 90.0f, 0.0f }, { -243.0f, 90.0f, 243.0f },
+    { 243.0f, 90.0f, -243.0f },  { 243.0f, 90.0f, 0.0f },  { 243.0f, 90.0f, 243.0f },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd/func_808CADC0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Boss_Fd/func_808CAE8C.s")
+static Vec3f sCeilingTargets[] = {
+    { 0.0f, 900.0f, -243.0f }, { 243.0, 900.0f, -100.0f },  { 243.0f, 900.0f, 100.0f },
+    { 0.0f, 900.0f, 243.0f },  { -243.0f, 900.0f, 100.0f }, { -243.0, 900.0f, -100.0f },
+};
+
+#ifdef NON_MATCHING
+// Somehow doesn't use rodata value D_808D1EB4 = 0.1f. It would occur after the 85.56f float
+// literal in case 6 of the boss intro switch statement but before the next switch statement.
+// All instructions match.
+
+void BossFd_Fly(BossFd* this, GlobalContext* globalCtx) {
+    u8 sp1CF = false;
+    u8 temp_rand;
+    s16 i1;
+    s16 i2;
+    s16 i3;
+    f32 dx;
+    f32 dy;
+    f32 dz;
+    Player* player = PLAYER;
+    f32 angleToTarget;
+    f32 pitchToTarget;
+    Vec3f* holePosition1;
+    f32 temp_y;
+    f32 temp_x;
+    f32 temp_z;
+    s32 pad19C;
+
+    SkelAnime_Update(&this->skelAnimeHead);
+    SkelAnime_Update(&this->skelAnimeRightArm);
+    SkelAnime_Update(&this->skelAnimeLeftArm);
+    dx = this->targetPosition.x - this->actor.world.pos.x;
+    dy = this->targetPosition.y - this->actor.world.pos.y;
+    dz = this->targetPosition.z - this->actor.world.pos.z;
+    dx += Math_SinS((2096.0f + this->fwork[BFD_FLY_WOBBLE_RATE]) * this->work[BFD_MOVE_TIMER]) *
+          this->fwork[BFD_FLY_WOBBLE_AMP];
+    dy += Math_SinS((1096.0f + this->fwork[BFD_FLY_WOBBLE_RATE]) * this->work[BFD_MOVE_TIMER]) *
+          this->fwork[BFD_FLY_WOBBLE_AMP];
+    dz += Math_SinS((1796.0f + this->fwork[BFD_FLY_WOBBLE_RATE]) * this->work[BFD_MOVE_TIMER]) *
+          this->fwork[BFD_FLY_WOBBLE_AMP];
+    angleToTarget = (s16)(Math_FAtan2F(dx, dz) * (0x8000 / M_PI));
+    pitchToTarget = (s16)(Math_FAtan2F(dy, sqrtf(SQ(dx) + SQ(dz))) * (0x8000 / M_PI));
+
+    osSyncPrintf("MODE %d\n", this->work[BFD_ACTION_STATE]);
+
+    Math_ApproachF(&this->fwork[BFD_BODY_PULSE], 0.1f, 1.0f, 0.02);
+
+    /****************************************************************************************
```

Comment:
Probably don't need this big of a comment

Reply:
would agree, you can use // comments like we did in dark link